### ISSUE TITLE
LogCapture.check_present() failure output is (sometimes) misleading

### ISSUE THREAD

**Issue by asqui** [Simplistix/testfixtures#136]
When `LogCapture.check_present()` is called with `order_matters=True` (the default) failure to find the expected log message amongst the captured entries gives an output in the form:

```
sequence not as expected:

same:
()

expected:
(FOO,)

actual:
(BAR, BAZ, FOB)
```

This can easily be mistaken for a standard `LogCapture.compare()` failure output, leading you astray wondering about where BAR and BAZ are coming from, when what you really should be looking at is why FOO != FOB.

Perhaps we can use a different `x_label` and `y_label` for the comparison in this branch to give a more obvious failure output?

**Comment by cjw296** [comment]
Sure, sounds like a good suggestion, can you throw in a PR?

**Comment by asqui** [comment]
Yep, the code change should be trivial. Is this too wordy?
```
sequence not as expected:

same:
()

expected to find the following (ordered) log messages...:
(FOO,)

...in the captured logging output:
(BAR, BAZ, FOB)
```

**Comment by cjw296** [comment]
Yes, too wordy :-)

A one to three word phrase for each should suffice...

**Comment by asqui** [comment]
If you could suggest your preferred wording I'll prepare a PR.

**Comment by cjw296** [comment]
```
same:
expected in this order:
actual:
```

You'll note that `.check()` doesn't do anything special with compare prefixes, and the traceback should show quite clearly that this is coming from a LogCapture.

**Comment by asqui** [comment]
Unfortunately an `x_label` of 'expected in this order' is not great because that label will be used recursively, and leads to output such as this (taken from `test_order_matters_not_okay_recursive()`):

```
sequence not as expected:

same:
()

expected in this order:
(('root', 'INFO', 'one'),)

actual:
(('root', 'ERROR', 'junk'),)

While comparing [0]: sequence not as expected:

same:
('root',)

expected in this order:
('INFO', 'one')

actual:
('ERROR', 'junk')

While comparing [0][1]: 'INFO' (expected in this order) != 'ERROR' (actual)
```

How about a prefix instead?

A prefix of "The expected ordered log entries were not present" gives the following output for the same test:
```
The expected ordered log entries were not present: sequence not as expected:

same:
()

expected:
(('root', 'INFO', 'one'),)

actual:
(('root', 'ERROR', 'junk'),)

While comparing [0]: sequence not as expected:

same:
('root',)

expected:
('INFO', 'one')

actual:
('ERROR', 'junk')

While comparing [0][1]: 'INFO' (expected) != 'ERROR' (actual)
```

(Open to suggestions for better wording.)

**Comment by cjw296** [comment]
That looks clunky, kinda surprised the next step up in the traceback doesn't give enough of a hint. 
I have a feeling the "sequence not as expected:" can be customised with some fiddling, but this feels like it's getting into "not worth changing" territory?

**Comment by asqui** [comment]
The real-world scenario I encountered was a failing test using `check_present()` which was indistinguishable from a `check()` failure based on the output alone.

The failure was on Jenkins rather than locally, so I had a stack trace and failure message, but the stack trace did not include a mention of `check_present()` because the expected log lines were split over multiple lines, and only the last line of the callsite is included in a standard Python traceback.

Maybe this is a nose/unittest-specific deficiency? I know pytest gives much more verbose output, which would probably have shown the complete call to `check_present()` in the failure output.

**Comment by cjw296** [comment]
Yeah, nose is way past EOL now. I'd suggest moving to pytest, it's become the defacto standard.
I'm going to close this for now, but if any obvious/easy solutions present themselves, ping a comment on and I'll re-open.
